paddle/fluid/framework/details/build_strategy.h

```diff
@@ -32,6 +32,8 @@ struct BuildStrategy {
   ReduceStrategy reduce_{ReduceStrategy::kAllReduce};
   GradientScaleStrategy gradient_scale_{GradientScaleStrategy::kCoeffNumDevice};
 
+  bool share_parameter_between_cards_{false};
```

Comment:
How to `share parameter between careds` when use cuda?

Reply:
Why we need another flag instead of `ReduceStrategy` ?

Reply:
> How to share parameter between cards when use cuda?

If `share_parameter_between_cards_` is true, `use_cuda_` must be false and `build_strategy.reduce_` must be `ReduceStrategy::kReduce`. There is a checking:
https://github.com/PaddlePaddle/Paddle/pull/11406/files#diff-564dec854cf4f37015001783f71e06cbR76

Reply:
So... Why we need the new flag rather than just use `build_strategy.reduce_` ?

The data fields should be ORTHOGONAL.